code/class/registry/registry.php

```diff
@@ -129,7 +136,54 @@ public function alias($class, $alias)
     }
 
     /**
-     * Get a list of all the identifier aliases
+     * Get a specific locator from the class namespace
+     *
+     * @return string|false The name of the locator or FALSE if no locator could be found
+     */
+    public function getLocator($class)
+    {
+        $result = false;
+
+        if(!false == $pos = strrpos($class, '\\'))
```

Comment:
The way this is implemented it wouldn't. The locator for the global namespace should be prepended to the list of locators so it always get called first. It improved upon this by ensuring the component locator is prepended.